live-build/misc/live-build-hooks/90-linux-migration-artifact.binary

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash -eux
+#
+# Copyright 2018 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This script is intended to be used as part of Delphix's build process.
+# Its role is to convert the "binary" directory generated by live-build,
+# into a cpio archive that will be wrapper in a tarball. That tarball is
+# the migration artifact that illumos-based engines use to migrate to
+# Linux.
+#
+MIGRATION_ARTIFACT="$APPLIANCE_VARIANT.migration.tar.gz"
+
+#
+# At the time of this writing dx_unpack.sh expects a tarball of the
+# following structure:
+#
+# depot/                       the top level container
+# depot/dx_prepare             top-level script to prepare the archive
+#                              for upgrade
+# depot/version.info           version and date information about the
+#                              upgrade image
+# depot/hashes                 the sha256sum of any upgrade artifacts
+#                              (see below)
+# depot/hashes.sig[.<release>] one cryptographic signature of depot/hashes
+#                              per Delphix release
+# depot/<upgrade artifacts>    generally cpio archives of OS and appliance
+#
+# As dx_upack.sh is the only piece of code that runs from the old
+# world, we treat it as a contract that lists the bare minimum
+# requirements for our artifact.
+#
+DEPOT_DIRECTORY=depot
+
+#
+# Delete any leftover files from previous runs.
+#
+rm -rf "$MIGRATION_ARTIFACT" "$DEPOT_DIRECTORY"
+
+#
+# Not all variants include the virtualization application (e.g.
+# internal-minimal) so don't bother generating a migration image
+# for those. /opt/delphix is always created as part of the
+# appliance. Thus, checking if it exists is a quick way for us to
+# distinguish between variants that have the appliance from
+# the ones that don't.
+#
+if [[ ! -d binary/opt/delphix ]]; then
+	echo "This artifact does not contain the appliance." \
+		" Skipping migration image..."
+	exit 0
+fi
+
+mkdir $DEPOT_DIRECTORY
+
+#
+# Generate archive for os root directory.
+#
+# Note: We temporarily change directory in the subshell (we move to the
+# "binary" directory) because find prints everything as a relative
+# path to the current directory and we don't want the generated archive
+# to have "binary" as its top-level directory.
+#
+(
+	cd binary
+	find ./* -print | cpio -oc
```

Comment:
`find ./*` will ignore all dot files directly under `./`. You should use `find .` instead